File: tests/Mono.Android-Tests/Mono.Android-Tests/System.IO/StreamTest.cs
using System;
using System.IO;
using NUnit.Framework;
using JavaStreamTest = global::Net.Dot.Android.Test.StreamTest;

namespace System.IOTests;

[TestFixture]
public class StreamTest
{
	[Test]
	public void InputStreamAdapter_Read ()
	{
		using var stream = new MemoryStream (new byte [JavaStreamTest.BufferSize]);
		var result = JavaStreamTest.InputStreamAdapter_Read (stream);
		Assert.AreEqual (0, result); // First byte is 0
	}

	[Test]
	public void InputStreamAdapter_Read_bytes ()
	{
		using var stream = new MemoryStream (new byte [JavaStreamTest.BufferSize]);
		var result = JavaStreamTest.InputStreamAdapter_Read_bytes (stream);
		Assert.AreEqual (JavaStreamTest.BufferSize, result);
	}

	[Test]
	public void InputStreamAdapter_Read_bytes_int_int ()
	{
		using var stream = new MemoryStream (new byte [JavaStreamTest.BufferSize]);
		var result = JavaStreamTest.InputStreamAdapter_Read_bytes_int_int (stream);
		Assert.AreEqual (JavaStreamTest.BufferSize, result);
	}
}

File: tests/Mono.Android-Tests/Mono.Android-Tests/java/net/dot/android/test/StreamTest.java
package net.dot.android.test;

import android.util.Log;
import java.io.InputStream;
import java.io.IOException;

public class StreamTest
{
    static final String TAG = "StreamTest";
    public static final int BUFFER_SIZE = 1024;

    public static int InputStreamAdapter_Read(InputStream stream) throws IOException
    {
        Log.d(TAG, "StreamTest.InputStreamAdapter_Read, underlying stream type: " + stream.getClass().getName());
        return stream.read();
    }

    public static int InputStreamAdapter_Read_bytes(InputStream stream) throws IOException
    {
        Log.d(TAG, "StreamTest.InputStreamAdapter_Read_bytes, underlying stream type: " + stream.getClass().getName());
        byte[] buffer = new byte[BUFFER_SIZE];
        return stream.read(buffer);
    }
    public static int InputStreamAdapter_Read_bytes_int_int(InputStream stream) throws IOException
    {
        Log.d(TAG, "StreamTest.InputStreamAdapter_Read_bytes_int_int, underlying stream type: " + stream.getClass().getName());
        byte[] buffer = new byte[BUFFER_SIZE];
        return stream.read(buffer, 0, BUFFER_SIZE);
    }
}
